### PR TITLE
Tenants: add create/update methods and maxUnits configuration

### PR DESCRIPTION
**Pull request by nicoloboschi**
* Added new endpoint for creating (POST) and updating a tenant (PATCH), similar to the application ones
* The PUT tenant is still there

The new commands can pass a new optional param `--max-units` (integer) (part of https://github.com/LangStream/langstream/issues/304)

The PUT endpoint doesn't accept any body. To create or update the tenant is recommended to use create/update from now on.

### CODE REVIEW

File: langstream-api/src/main/java/ai/langstream/api/webservice/tenant/CreateTenantRequest.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class CreateTenantRequest {
+
+    private Integer maxUnits;
```
Comment (eolivelli) [inline langstream-api/src/main/java/ai/langstream/api/webservice/tenant/CreateTenantRequest.java:14]: maxTotalResourceUnits

File: langstream-api/src/main/java/ai/langstream/api/webservice/tenant/TenantConfiguration.java [hunk 2]
```diff
@@ -26,4 +26,5 @@
 @AllArgsConstructor
 public class TenantConfiguration {
     private String name;
+    private int maxUnits;
```
Comment (eolivelli) [inline langstream-api/src/main/java/ai/langstream/api/webservice/tenant/TenantConfiguration.java:29]: maxTotalResourceUnits

File: langstream-api/src/main/java/ai/langstream/api/webservice/tenant/UpdateTenantRequest.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class UpdateTenantRequest {
+
+    private Integer maxUnits;
```
Comment (eolivelli) [inline langstream-api/src/main/java/ai/langstream/api/webservice/tenant/UpdateTenantRequest.java:14]: maxTotalResourceUnits

File: langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/CreateTenantCmd.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        name = "create",
+        mixinStandardHelpOptions = true,
+        description = "Create a new tenant")
+public class CreateTenantCmd extends BaseTenantCmd {
+
+    @CommandLine.Parameters(description = "Name of the tenant")
+    private String name;
+
+    @CommandLine.Option(
+            names = {"--max-units"},
```
Comment (eolivelli) [inline langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/CreateTenantCmd.java:33]: --max-total-resource-units

File: langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/UpdateTenantCmd.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        name = "update",
+        mixinStandardHelpOptions = true,
+        description = "Update an existing tenant")
+public class UpdateTenantCmd extends BaseTenantCmd {
+
+    @CommandLine.Parameters(description = "Name of the tenant")
+    private String name;
+
+    @CommandLine.Option(
+            names = {"--max-units"},
```
Comment (eolivelli) [inline langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/UpdateTenantCmd.java:33]: --max-total-resource-units

File: langstream-webservice/src/main/java/ai/langstream/webservice/common/TenantResource.java [hunk 6]
```diff
@@ -62,6 +69,42 @@ TenantConfiguration getTenant(@NotBlank @PathVariable("tenant") String tenant) {
...
     }
 
+    @PostMapping(value = "/{tenant}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Create a tenant")
+    void createTenant(
+            @NotBlank @PathVariable("tenant") String tenant,
+            @RequestBody CreateTenantRequest createTenantRequest)
+            throws TenantException {
+        try {
+            globalMetadataService.createTenant(tenant, createTenantRequest);
```
Comment (eolivelli) [inline langstream-webservice/src/main/java/ai/langstream/webservice/common/TenantResource.java:79]: we need an hard limit for the "--max-total-resource-units", configurable in the control plane

**Review by eolivelli — CHANGES_REQUESTED** [review]
nice work!

I have left some comments, PTAL

**Comment by nicoloboschi** [comment]
@eolivelli thanks, I renamed the property and added a configurable limit in the control plane config. PTAL again

p.s. I'll do the rename of the deployer/control-plane later

**Comment by nicoloboschi** [comment]
Merged as part of https://github.com/LangStream/langstream/pull/326

**Pull request closed without merging**